src/main/java/org/beanpod/switchboard/repository/StreamStatRepository.java

```diff
@@ -11,7 +12,11 @@
 
   StreamStatEntity save(StreamStatEntity streamStatEntity);
 
-  Optional<StreamStatEntity> findStreamStatEntityById(Long id);
+  Optional<StreamStatEntity>
+      findByStreamInputChannelDecoderDeviceUserAndIdOrStreamOutputChannelEncoderDeviceUserAndId(
+          UserEntity inputUser, Long inputId, UserEntity outputUser, Long outputId);
```

Comment:
really confused about these or statements, how does this equate to finding a specific stream stat corresponding to the user

Reply:
maybe we can put a a comment above like
//finds stream stats associated with a user
so we directly know what it does

Reply:
If you're getting confused, you might need to brush up on set theory and the precedence of logic operators (NOT, AND, OR) 😛. JPA Data Method Queries can get quite verbose for the Spring Data parser. I originally thought about adding a comment, but the method name is self-explanatory and the naming convention can be refrenced in the [Spring Data Docs](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details).